Mathlib/RingTheory/MvPolynomial/Symmetric.lean

```diff
@@ -298,32 +350,127 @@ theorem degrees_esymm [Nontrivial R] (n : ℕ) (hpos : 0 < n) (hn : n ≤ Fintyp
 
 end ElementarySymmetric
 
+section CompleteHomogeneousSymmetric
+
+open Finset Multiset Sym
+
+variable [DecidableEq σ] [DecidableEq τ] (n : ℕ)
+
+/-- The `n`th complete homogeneous symmetric `MvPolynomial σ R`. -/
+def hsymm : MvPolynomial σ R := ∑ s : Sym σ n, (s.1.map X).prod
+
+lemma hsymm_def : hsymm σ R n = ∑ s : Sym σ n, (s.1.map X).prod := rfl
+
+/-- `hsymmMu` is the product of the symmetric polynomials `hsymm μᵢ`,
+where `μ = (μ₁, μ₂, ...)` is a partition. -/
+def hsymmMu {n : ℕ} (μ : n.Partition) : MvPolynomial σ R := muProduct σ R (hsymm σ R) μ
```

Comment:
I don't quite like how you name `hsymmMu` after the free variable `μ`. Is this really standard in the literature?

Reply:
I don't like it either, and It's really not standard. The standard notation would be just h_\mu. In the literature there is no distinction (in notation) between h_n and h_\mu, so I would really like to call it just hsymm, but the name is taken. I'm very much open to suggestions.

Reply:
Maybe `hsymmPartition` or `hsymmPart`?

Reply:
Sounds good. Do you have a similar suggestion for `muProduct`?

Reply:
Honestly, I'm not sure we need `muProduct` as a def? Like, writing `(μ.parts.map f).prod` isn't so bad.

Reply:
Yeah you're right. I got rid of it.